modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/H2IndexCostedBase.java

```diff
@@ -641,22 +664,558 @@ else if (needsToReadFromScanIndex)
          * Cost function.
          * See more: {@link BaseIndex#getCostRangeIndex}.
          */
-        long getCostRangeIndex(int[] masks, long rowCount,
-            TableFilter[] filters, int filter, SortOrder sortOrder,
-            boolean isScanIndex, AllColumnsForPlan allColumnsSet);
+        long getCostRangeIndex(Session ses, int[] masks, long rowCount,
```

Comment:
let's get rid of the `rowCount` param, since we could get this by ` getRowCountApproximation(ses);` where it is actually needed